admin/client/EditorBasicTab.tsx

```diff
@@ -199,16 +200,30 @@ export class EditorBasicTab extends React.Component<{ editor: ChartEditor }> {
                         options={ChartTypeDefs.map(def => def.key)}
                         optionLabels={ChartTypeDefs.map(def => def.label)}
                     />
+                    {editor.features.explorer && (
+                        <FieldsRow>
+                            <Toggle
+                                label="Explorable chart"
+                                value={chart.props.isExplorable}
+                                onValue={value =>
+                                    (chart.props.isExplorable = value)
+                                }
+                                disabled={!canBeExplorable(chart.props)}
```

Comment:
Why is this a utility function, instead of a method on the chart object?

Reply:
So my thought process was:
- `ChartConfigProps` doesn't have any methods and the convention is to treat it as a dumb object that has no behaviour, only fields.
- The constructor on `ChartConfig` initializes a whole bunch of stuff, including data loading, so having it there is not an option if we want to check whether a chart is explorable on the back-end.
- Checking for `isExplorable` on the back-end is important in case we change the logic, e.g. we exclude bar charts. Then, having this check on the back-end will make sure a previously created bar chart doesn't get published as an explorable chart. The alternative is that we migrate the config when we make a change to the logic, which is more error-prone.

Do you think it's better to turn `ChartConfigProps` into an object with behaviour?

Reply:
What's the motivation for making it a dumb object with no behavior? I generally feel it's better to put behavior on the object rather than to have a random separate file with utility functions that apply to that one object.

Needing the logic on the back-end makes sense to me, would be nice if there were a more elegant way to do it, but this seems fine

Reply:
The `class ChartConfigProps` should really be an `interface` since we are **always** treating it as such and it's a useful no-behaviour thing to have. We can then have an object with behaviour that accepts `ChartConfigProps` and is used to set & get props. And then finally the `ChartConfig` that has all the initialization behaviour – fetching the data, passing it to all the objects, etc.

I would be in favour of re-architecting it that way but I think it's maybe not a trivial amount of work.

Reply:
Maybe it's not too involved, we just need to move all the types to an `interface` and have the class `implements` that interface. Then it's a question of whether mobx `@observable` would have an identical behaviour as it does now applied to individual properties. Still, it's some work to verify and test that...

Reply:
Cool. Up to you if you want to do that work now or leave it for another PR, I'm good with merging this

Reply:
I think it's better to leave as a refactoring task, wouldn't like to dive into that now.